### DISCUSSION TITLE
Creating a dictionary for the LaSalle/TAO theory (French)

### DISCUSSION THREAD

**Discussion by Vermoot** [opensteno/plover#1427] [Systems and languages]
Hello!

I've been trying out Plover in English and liking it very much. Now I'd love to be able to use it for French. Having spent a bit of time on the Plover Discord, I think I'm not the only one with interest in the matter.

I know French theories/dictionaries have been discussed before, but from [what I've seen](https://github.com/openstenoproject/plover/wiki/Steno-Layouts-&-Supported-Languages#french) there's support for the Grandjean theory, but that theory isn't very realtime-friendly. The fact it doesn't use the same layout as the "standard" English steno machines (and the Plover theory) isn't ideal either.

So now there's TAO (Transcription Assistée par Ordinateur, also often called LaSalle) theory, which is both using the standard STPH keyboard *and* made to be realtime-friendly.

There's a book going around that aims to teach TAO, which is obviously an amazing resource. There's theory, briefs, etc. It's a scan, but seems to have been OCR'd or something, allowing us to search through it, and even potentially get data out of it to help create a dictionary. The legality of this is a bit fuzzy to me, but talking about it on Discord people seemed to think this was probably mostly okay.

[@morinted even started turning this book into a dictionary already](https://github.com/morinted/la_tao_dictionary)!! Although the repo hasn't seen any activity in years, and the dictionary isn't enough to start steno-ing in French effectively (especially for beginners learning the theory), I believe that could be a great jump start.

Now I'm not a theory wizard, I'm no genius dev etc, but this issue aims to start a common effort to create a viable dictionary for TAO/LaSalle, which seems to be a great option for French stenography. @jenchanws has hinted at the possibility of "autogenerating" a dictionary, however I have no idea how one would do that.

Any input would be appreciated, I really do believe we can do something cool with TAO, effectively adding French to the list of usable languages in Plover. And wouldn't that be cool?

Cheers.

**Comment by Vermoot** [comment]
Alright so after a bit of discussion on Discord, here's where we are:

Using the book as a guide to create a dictionary should be alright, and then making the dictionary public but writing new learning material (probably in English and in French) seems to be the way to go. That way we don't re-publish, copy, translate (...) copyrighted material.

So now I've started compiling the rules in the guide and translating them to English so that other people can use them to create a script to generate a dictionary, following the theory's rules. I guess this would be done using a French IPA dictionary, or something like that.

Here's an excerpt of what that will/could look like:

> ### Lesson 1 : Initial S, initial T, final S, final T, vowels A O E U
> 
> 
> RULE:  
> We don't write the final `e` because it's silent. Examples: tasse TAS, passe PAS, casse KAS, sorte SORT, porte PORT, croûte KROUT, crise KRIZ.
> 
> 
> The -S is also read R to form the sound `tre` (like in the word "être").
> 
> 
> ### Lesson 2 : initial P and final PORT
> 
> No rule.
> 
> ### Lesson 3 : Initial and final R
> 
> Vowel: the E designates the sound "é"
> 
> ### Lesson 4 : Initial and final V
> 
> RULE:  
> The W key designates the initial V sound  
> The F key designates the final V sound
> 
> Vowel: Keys OE designate the "long O" sound (sauve SOEF, sauce SOES, taupe TOEP)
> 
> RULE:  
> The star in a chord is used to distinguish verbs and nouns that are homophones. Starred chords are verbs. (TODO: Verify that. Is it always the case? The examples in this particular lesson aren't very telling.)

I'm not at all focusing on layout or anything like that. This is not meant to become documentation nor learning material; only a guide for anyone working on a script. @EPLHREU has expressed a willingness to help with that, but of course there's no pressure. Anyone else who's interested is welcome to join.

**Reply by panathea** [reply]
This seems great, any progress on mapping out the book into a set of rules?

**Reply by Vermoot** [reply]
Hi! Yes!

I've started a whole new project called [Pluvier](https://github.com/Vermoot/Pluvier) aiming to create the dictionary generator, exactly how I said in this post.

I did compile the whole book into a set of rules, and then organised them into a set of phonetic rules and a set of orthographic rules. I even started transcribing them into python, without really knowing what I'd do with them in the end, but I figured it was a good start.

There's still lots to do, and the transcribing into python isn't done by any means (I have a separate rules files in which I delete ones I've already done), but I've put it aside for a little while, because I'm working on other things (and I'm not the best at keeping focused on projects like that :D). Also, being unsure of the direction I should be taking didn't help.

If you're interested in working on this or giving me pointers, I'd be very happy with that. You can also find me on the Plover discord, I've created a Pluvier thread with details of my advancement.
